src/ParseServer.js

```diff
@@ -376,6 +376,38 @@ function injectDefaults(options: ParseServerOptions) {
     options.serverURL = `http://localhost:${options.port}${options.mountPath}`;
   }
 
+  if (options.appId) {
+    const reservedCharacters = [
+      '!',
+      '#',
+      '$',
+      '%',
+      '&',
+      "'",
+      '(',
+      ')',
+      '*',
+      '+',
+      '/',
+      ':',
+      ';',
+      '=',
+      '?',
+      '@',
+      '[',
+      ']',
+    ];
+    if (
+      reservedCharacters.some(function(v) {
+        return options.appId.indexOf(v) >= 0;
```

Comment:
You can try something like this (I added a few more characters)

```
// Reserved Characters
const regex = /[!#$%'()*+/:;=?@[\]{}^,|<>]/g
if (options.appId.match(regex) {
// log here
}
```

Reply:
For anybody reading this `/[...]/` means match any character inside the brackets. The only escaped character in this regex is the left bracket. `/[...[\]...]/`